sirepo/job_api.py

```diff
@@ -100,7 +108,14 @@ async def api_downloadDataFile(
                     req_data=req.req_data,
                     suffix=s,
                 )
-                assert not r.state == "error", f"error state in request=={r}"
+                if r.state == sirepo.job.CANCELED:
+                    # POSIT: Users can't cancel donwloadDataFile. So canceled means there was a
+                    # timeout (max_run_secs exceeded).
+                    _raise_content_too_large(req)
+                if r.state == sirepo.job.ERROR:
+                    if r.get("errorCode") == sirepo.job.ERROR_CODE_RESPONSE_TOO_LARGE:
```

Comment:
This is fine, but it is also problematic, because every place error is checked, we have to check errorCode. Ideally, we would centralize this type of error handling. For now, it's "fine".